src/coreclr/jit/valuenum.h

```diff
@@ -206,6 +206,21 @@ struct VNFuncApp
     }
 };
 
+struct VNPhiDef
+{
+    unsigned  LclNum;
+    unsigned  SsaDef;
+    unsigned* SsaArgs;
+    unsigned  NumArgs;
+};
+
+struct VNMemoryPhiDef
+{
+    BasicBlock* Block;
+    unsigned*   SsaArgs;
+    unsigned    NumArgs;
+};
```

Comment:
Hmm, it would require introducing some "kind" type in the unified struct anyway that you will need to check. To actually use the SSA args you need to differentiate between whether it's memory or local, so I'm not sure it simplifies much. I think I'll leave it like this and then we can introduce a helper if there is a good way to unify it.